### PR TITLE
TT-2133: fix CORS pass through for GraphQL

### PR DESCRIPTION
**Pull request by urbanishimwe**
The error was probably caused by the missing GQL request object in a request context,
we fix this by checking if the error is of that type and fallback by sending a normal HTTP request!

## Motivation and Context
https://tyktech.atlassian.net/browse/TT-2133

## How This Has Been Tested
create a GraphQL proxy only API and send a pre-flight request and check if the request has reached upstream successfully

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

### CODE REVIEW

File: gateway/testutil.go [hunk 1]
```diff
@@ -939,7 +940,8 @@ func (s *Test) Start() {
 		scheme = "https://"
 	}
 	s.URL = scheme + mainProxy().listener.Addr().String()
-
+	// set simple CORS options
+	controlProxy().router.Use(cors.New(cors.Options{}).Handler)
```
Comment (spetrunin) [inline gateway/testutil.go:944]: @urbanishimwe why this change is required?

Comment (urbanishimwe) [reply gateway/testutil.go:944]: for the server to handle preflight request

Comment (spetrunin) [reply gateway/testutil.go:944]: I've commented this line - and tests are still working

Comment (urbanishimwe) [reply gateway/testutil.go]: me too! removing it

**Review by pvormste — APPROVED** [review]

**Review by spetrunin — APPROVED** [review]
